keps/sig-node/1867-disable-accelerator-usage-metrics/README.md

```diff
@@ -0,0 +1,177 @@
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [X] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements](https://github.com/kubernetes/enhancements/issues/1867)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [X] (R) Design details are appropriately documented
+- [X] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [X] (R) Graduation criteria is in place
+- [X] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP outlines the process to deprecate the Accelerator Metrics collected by Kubelet.
+
+Accelerator metrics are no longer expected to collected by Kubelet, but by external monitoring agents using the PodResources API.
+The purpose of creating that API was to provide an out of tree mechanism for all vendors to provide device specific metrics.
+This allows them to provide these metrics without requiring them to make changes to Kubernetes.
+
+Now that this API is beta and soon to be G.A, this KEP outlines the process to deprecate metrics that were added before sig-node conveged on the PodResources API.
+
+## Motivation
+
+### Goals
+
+Deprecate and remove the AcceleratorUsage metrics that kubelet currently advertises.
+
+### Non-Goals
+
+Deprecation and removal of that API is certainly a goal of sig-node, and the step of removing the AcceleratorUsage metric is a step in that direction (agreed upon goal with incremental steps towards it) but not the goal of this KEP.
+
+## Proposal
+
+### Risks and Mitigations
+
+The main risk that we face is breaking existing consumers of the AcceleratorUsage metrics.
+The way this risk is mitigated is through the use of Feature Flags, which allows users to re-enable this metric.
+We will also be pointing in documentation users towards the newer and richer method of collecting metrics.
+
+Note that we don't know who are the consumers of that metric but we suspect that this will impact a small subset of users as these metrics on NVIDIA GPU utilization are often unreliable and very coarse.
+
+## Design Details
+
+Add a feature flag and pass the disable option to cadvisor.
+
+### Test Plan
+
+* E2E tests that checks when the feature flag is enabled if the metrics are present or not.
+
+### Graduation Criteria
+
+#### Alpha Graduation
+* Feature Flag is present.
```

Comment:
The awkward part of this process is that enabling this feature flag is recommended by all deployers at this phase.  I think we should document and promote immediate usage of this flag even if it’s alpha.

@brancz has sig-monitoring encountered a similar scenario?  Usage of these metrics actually impairs the ability to consume what is measured.

Reply:
A situation exactly like this hasn't come up, but generally speaking, since we have the metrics stability framework in place and all these metrics are alpha metrics, theoretically by the metrics stability framework you wouldn't even have to have a feature flag, but you could just remove them.